test/cli/test/scoped-fixtures.test.ts

```diff
@@ -0,0 +1,771 @@
+    'basic.test.ts': () => {
+      const extendedTest = it.extend<{
+        file: string
+        local: string
+      }>({
+        local: 'local',
+        file: [
+          ({ local }, use) => use(local),
+          { scope: 'file' },
+        ],
+      })
+
+      extendedTest('working', ({ file, local }) => {
+        expect(file).toBe(local)
+        expect(file).toBe('local')
+      })
+    },
+    'vitest.config.js': { test: { globals: true } },
+  })
+  expect(stdout).toContain('basic.test.ts')
+  expect(stderr).toBe('')
+})
+
+test('worker fixture works in vmThreads and runs for every file', async () => {
+  const { stderr, fixtures, tests } = await runFixtureTests(({ log }) => it.extend<{ worker: string }>({
+    worker: [
+      async ({}, use) => {
+        log('init worker')
+        await use('worker')
+        log('teardown worker')
+      },
+      { scope: 'worker' },
+    ],
+  }), {
+    '1-basic.test.ts': ({ extendedTest }) => {
+      extendedTest('test1', ({ worker: _worker }) => {})
+    },
+    '2-basic.test.ts': ({ extendedTest }) => {
+      extendedTest('test1', ({ worker: _worker }) => {})
+    },
+    'vitest.config.ts': {
+      test: {
+        globals: true,
+        maxWorkers: 1,
+        minWorkers: 1,
+        pool: 'vmThreads',
+      },
+    },
+  })
+
+  expect(stderr).toBe('')
+  expect(fixtures).toMatchInlineSnapshot(`
+    ">> fixture | init worker | test1
+    >> fixture | teardown worker | test1
+    >> fixture | init worker | test1
+    >> fixture | teardown worker | test1"
+  `)
+  expect(tests).toMatchInlineSnapshot(`
+    " ✓ 1-basic.test.ts > test1 <time>
+     ✓ 2-basic.test.ts > test1 <time>"
+  `)
+})
+
+test('worker fixtures in isolated tests init and teardown twice', async () => {
+  const { stderr, fixtures, tests } = await runFixtureTests(({ log }) => it.extend<{ worker: string }>({
+    worker: [
+      async ({}, use) => {
+        log('init worker')
+        await use('worker')
+        log('teardown worker')
+      },
+      { scope: 'worker' },
+    ],
+  }), {
+    '1-basic.test.ts': ({ extendedTest }) => {
+      extendedTest('test1', ({ worker: _worker }) => {})
+    },
+    '2-basic.test.ts': ({ extendedTest }) => {
+      extendedTest('test1', ({ worker: _worker }) => {})
+    },
+    'vitest.config.ts': {
+      test: {
+        globals: true,
+        isolate: true,
+        maxWorkers: 2,
+        minWorkers: 2,
+        pool: 'threads',
+      },
+    },
+  })
+
+  expect(stderr).toBe('')
+  expect(fixtures).toMatchInlineSnapshot(`
+    ">> fixture | init worker | test1
+    >> fixture | teardown worker | test1
+    >> fixture | init worker | test1
+    >> fixture | teardown worker | test1"
+  `)
+  expect(tests).toMatchInlineSnapshot(`
+    " ✓ 1-basic.test.ts > test1 <time>
+     ✓ 2-basic.test.ts > test1 <time>"
+  `)
+})
+
+// TODO: no-isolate breaks tests
+test.skip('worker fixture initiates and torn down in different workers', async () => {
```

Comment:
This is breaking since message ports that are used to communicate in rpc channel are closed before teardown is called. 

- When `isolate: false` is used, teardown is called when `pool.destroy()` is called. 
- When `isolate: true` is called, teardown is called when `pool.run()` resolves.

Rpc cannot send messages after this part:

https://github.com/vitest-dev/vitest/blob/c2106b337b386773f6e6a9f0a64c8049cc540c37/packages/vitest/src/node/pools/threads.ts#L147-L150